components/suggest/src/rs.rs

```diff
@@ -58,8 +58,18 @@ pub(crate) const DEFAULT_RECORDS_TYPES: [SuggestRecordType; 9] = [
 ///
 /// This trait lets tests use a mock client.
 pub(crate) trait Client {
-    /// Fetch a list of records and attachment data
-    fn get_records(&self, request: RecordRequest) -> Result<Vec<Record>>;
+    /// Get all records from the server
+    ///
+    /// We use this plus client-side filtering rather than any server-side filtering, as
+    /// recommended by the remote settings docs
+    /// (https://remote-settings.readthedocs.io/en/stable/client-specifications.html). This is
+    /// relatively inexpensive since we use a cache and don't fetch attachments until after the
+    /// client-side filtering.
+    ///
+    /// Records that can't be parsed as [SuggestRecord] are ignored.
+    fn get_records(&self, db: &SuggestDb, fetch_fakespot_records: bool) -> Result<Vec<Record>>;
```

Comment:
I tried out letting the consumer specify the collection and not special-casing fakespot in general and I like the new codebetter.

It's not so easy to remove the dependency on a DB (now a `SuggestDao`), since it also sometimes writes to the DB.  It's definitely possible, but I'm kind of hoping to avoid it since all of this code is just a stop-gap until #6328 is implemented.  Once that happens, then the remote settings client will handle caching internally and we can drop the argument.